Geometry_Engine/Modify/SplitAtPoints.cs

```diff
@@ -212,111 +212,113 @@ public static List<PolyCurve> SplitAtPoints(this PolyCurve curve, List<Point> po
                     {
-                        if (point.IIsOnCurve(crv, tolerance))
-                            subPoints.Add(point);
-                    }
-                    tmpResult.AddRange((crv as Line).SplitAtPoints(subPoints));
-                    subPoints.Clear();
-                }
-                else if (crv is Circle)
-                {
-                    List<PolyCurve> tResult = new List<PolyCurve>();
-                    foreach (Arc arc in (crv as Circle).SplitAtPoints(onCurvePoints, tolerance))
-                        tResult.Add(new PolyCurve { Curves = new List<ICurve> { arc } });
-                    result.AddRange(tResult);
-                }
-                else
-                {
-                    Base.Compute.RecordError($"SplitAtPoints is not implemented for PolyCurves consisting of ICurves of type: {crv.GetType().Name}.");
-                    return null;
-                }
-            }
-
-            int i = 0;
-            int j = 0;
-
-            if (curve.IStartPoint().IsEqual(onCurvePoints[0]))
-            {
-                onCurvePoints.Add(onCurvePoints[0]);
-                onCurvePoints.RemoveAt(0);
-            }
+                        if (ptsOnSegment.Any(x => x.SquareDistance(crv.IStartPoint()) < sqTol))  //Split at start
+                        {
+                            if (prev != null)
+                                result.Add(prev);   //Add prev to return list
 
-            bool isClosedAndAntiClockwise = curve.IIsClosed(tolerance) && !curve.IIsClockwise(curve.Normal(), tolerance);
+                            prev = new PolyCurve { Curves = new List<ICurve> { split[0] } };    //Set prev to current segment
+                        }
+                        else    //Split at end
+                        {
+                            if (prev != null)   //Prev set
+                            {
+                                prev.Curves.Add(split[0]);  //Add full split segment to prev
+                                result.Add(prev);   //Add prev to output list
+                            }
+                            else    //Prev not set
+                            {
+                                result.Add(new PolyCurve { Curves = new List<ICurve> { split[0] } });   //Add full split segment to output list
+                            }
+                            prev = null;    //Split at end -> set prev to null as nothing to bring over to next segment
+                        }
+                    }
+                    else    //More than one split segment
+                    {
+                        if (prev != null)   //Prev contains parts from previous segment, unless split at endpoint
+                        {
+                            Point stPt = crv.IStartPoint();
+                            if (ptsOnSegment.Any(x => x.SquareDistance(stPt) < sqTol))  //If split at startpoint
```

Comment:
Not sure if faster, but I would consider switching to the suggestion below (also another step to get rid of `ptsOnSegment`, which I suggested not to extract a few comments ago)
```suggestion
                            if (split[0].IStartPoint().SquareDistance(stPt) < sqTol)  //If split at startpoint
```

Reply:
Right, good point. Me stupid 🤦‍♂️